include/manif/impl/sgal3/SGal3Tangent_base.h

```diff
@@ -0,0 +1,635 @@
+   *    L:  (32)
+   *    M:  (33) = Q(nu,theta)
+   *    N:  (34) = N1 - N2
+   *    N1: (35) = Q(rho,theta)
+   *    N2: (36)
+   *
+   * Note that we use the following temporary blocks for computation
+   *
+   * Jl = [ .  .  E   .
+   *        V  .  .  rho
+   *        W W^2 . theta
+   *        .  .  .   .   ]
+   *
+   */
+
+  Jacobian Jl;
+
+  // theta vector
+  const Eigen::Map<const SO3Tangent<Scalar>> so3 = asSO3();
+
+  // Skew matrix W = theta^
+  Jl.template block<3, 3>(3, 0) = so3.hat();
+  ConstRef33 W = Jl.template block<3, 3>(3, 0);
+
+  // Skew matrix W^2
+  Jl.template block<3, 3>(6, 3) = W * W;
+  ConstRef33 WW = Jl.template block<3, 3>(6, 3);
+
+  // Skew matrix V = nu^
+  Jl.template block<3, 3>(6, 0) = skew(lin2());
+  ConstRef33 V = Jl.template block<3, 3>(6, 0);
+
+  // Angles and trigonometry
+  const Scalar theta_sq = so3.coeffs().squaredNorm();
+  // rotation angle
+  const Scalar theta = sqrt(theta_sq);
+  const Scalar theta_cu = theta * theta_sq;
+
+  const Scalar sin_t = sin(theta);
+  const Scalar cos_t = cos(theta);
+
+  // Blocks D
+  Jl.template topLeftCorner<3, 3>() = so3.ljac();
+  Jl.template block<3, 3>(3, 3) = Jl.template topLeftCorner<3, 3>();
+  Jl.template block<3, 3>(6, 6) = Jl.template topLeftCorner<3, 3>();
+
+  // Block E
+  // Note - we use here a temporary block to hold E
+  Jl.template block<3, 3>(0, 6) = I33(Scalar(0.5));
+  if (theta_sq > Constants<Scalar>::eps) {
+    const Scalar A = (theta - sin_t) / theta_sq / theta;
+    const Scalar B = (
+      theta_sq + Scalar(2) * cos_t - Scalar(2)
+    ) / (Scalar(2) * theta_sq * theta_sq);
+
+    Jl.template block<3, 3>(0, 6).noalias() += A * W + B * WW;
+  }
+
+  // Block E * nu
+  Jl.template block<3, 1>(0, 9) = Jl.template block<3, 3>(0, 6) * lin2();
+
+  // Block L
+  Scalar cA, cB;
+  // small angle approx.
+  if (theta_cu > Constants<Scalar>::eps) {
+    cA = (sin_t - theta * cos_t) / theta_cu;
+    cB = (
+      theta_sq + Scalar(2) * (Scalar(1) - theta * sin_t - cos_t)
+    ) / (Scalar(2) * theta_sq * theta_sq);
+  } else {
+    cA = Scalar(1./3.)  - Scalar(1./30.) * theta_sq;
+    cB = Scalar(1./8.);
+  }
+
+  // Block - L * t
+  Jl.template block<3, 3>(0, 3).noalias() = -t() * (
+    // Block L
+    I33(Scalar(0.5)) + cA * W + cB * WW
+  );
+
+  // Block M = Q(nu, theta)
+  SE3Tangent<Scalar>::fillQ(
+    Jl.template block<3, 3>(3, 6), coeffs().template segment<6>(3)
+  );
+
+  // Block N1, part of N. N1 = Q(rho, theta)
+  Jl.template block<6, 1>(3, 9) << lin(), ang();
+  ConstRef61 rho_theta = Jl.template block<6, 1>(3, 9);
+  // block N1 = Q(rho,theta)
+  SE3Tangent<Scalar>::fillQ(Jl.template block<3, 3>(0, 6), rho_theta);
+
+  // Block N2, part of N
+  Scalar cC, cD, cE, cF;
+  if (theta_cu > Constants<Scalar>::eps) {
+    cA = (Scalar(2) - theta * sin_t - Scalar(2) * cos_t) / theta_cu / theta;
+    cB = (
+      theta_cu + Scalar(6) * theta + Scalar(6) * theta * cos_t - Scalar(12) * sin_t
+    ) / (Scalar(6) * theta_cu * theta_sq);
```

Comment:
do we want to create pre-multiplied variables theta_four, theta_five? I leave it to you. So far we have theta_sq and theta_cu

Reply:
theta_six is not necessary as it only appears once. But a `const` definition comes at no cpu cost and it's cleaner code.

Reply:
Let's leave that to a potential sub-sequent pr